common-service/src/main/java/org/sunbird/global/ServiceBaseGlobal.java

```diff
@@ -0,0 +1,271 @@
+    if (StringUtils.isBlank(channel)) {
+      channel = JsonKey.DEFAULT_ROOT_ORG_ID;
+    }
+    reqContext.put(JsonKey.CHANNEL, channel);
+    ctx.flash().put(JsonKey.CHANNEL, channel);
+    reqContext.put(JsonKey.ENV, getOperationEnv(request));
+    reqContext.put(JsonKey.REQUEST_ID, ExecutionContext.getRequestId());
+
+    if (!USER_UNAUTH_STATES.contains(userId)) {
+      reqContext.put(JsonKey.ACTOR_ID, userId);
+      reqContext.put(JsonKey.ACTOR_TYPE, JsonKey.USER);
+      ctx.flash().put(JsonKey.ACTOR_ID, userId);
+      ctx.flash().put(JsonKey.ACTOR_TYPE, JsonKey.USER);
+    } else {
+      String consumerId = request.getHeader(HeaderParam.X_Consumer_ID.getName());
+      if (StringUtils.isBlank(consumerId)) {
+        consumerId = JsonKey.DEFAULT_CONSUMER_ID;
+      }
+      reqContext.put(JsonKey.ACTOR_ID, consumerId);
+      reqContext.put(JsonKey.ACTOR_TYPE, JsonKey.CONSUMER);
+      ctx.flash().put(JsonKey.ACTOR_ID, consumerId);
+      ctx.flash().put(JsonKey.ACTOR_TYPE, JsonKey.CONSUMER);
+    }
+
+    context.setRequestContext(reqContext);
+    Map<String, Object> map = new HashMap<>();
+    map.put(JsonKey.CONTEXT, TelemetryUtil.getTelemetryContext());
+    Map<String, Object> additionalInfo = new HashMap<>();
+    additionalInfo.put(JsonKey.URL, url);
+    additionalInfo.put(JsonKey.METHOD, methodName);
+    additionalInfo.put(JsonKey.START_TIME, startTime);
+
+    // additional info contains info other than context info ...
+    map.put(JsonKey.ADDITIONAL_INFO, additionalInfo);
+    if (StringUtils.isBlank(messageId)) {
+      messageId = JsonKey.DEFAULT_CONSUMER_ID;
+    }
+    ctx.flash().put(JsonKey.REQUEST_ID, messageId);
+    requestInfo.put(messageId, map);
+  }
+
+  private String getEnv(Request request) {
+
+    String uri = request.uri();
+    String env;
+    if (uri.startsWith("/v1/user")) {
+      env = JsonKey.USER;
+    } else if (uri.startsWith("/v1/org")) {
+      env = JsonKey.ORGANISATION;
+    } else if (uri.startsWith("/v1/object")) {
+      env = JsonKey.ANNOUNCEMENT;
+    } else if (uri.startsWith("/v1/page")) {
+      env = JsonKey.PAGE;
+    } else if (uri.startsWith("/v1/course/batch")) {
+      env = JsonKey.BATCH;
+    } else if (uri.startsWith("/v1/notification")) {
+      env = JsonKey.NOTIFICATION;
+    } else if (uri.startsWith("/v1/dashboard")) {
+      env = JsonKey.DASHBOARD;
+    } else if (uri.startsWith("/v1/badges")) {
+      env = JsonKey.BADGES;
+    } else if (uri.startsWith("/v1/issuer")) {
+      env = BadgingJsonKey.BADGES;
+    } else if (uri.startsWith("/v1/content")) {
+      env = JsonKey.BATCH;
+    } else if (uri.startsWith("/v1/role")) {
+      env = JsonKey.ROLE;
+    } else if (uri.startsWith("/v1/note")) {
+      env = JsonKey.NOTE;
+    } else if (uri.startsWith("/v1/location")) {
+      env = JsonKey.LOCATION;
+    } else {
+      env = "miscellaneous";
+    }
+    return env;
+  }
+
+  /**
+   * This method will do request data validation .
```

Comment:
Returns a promise with error result in case of validation error.

request HTTP request in play context
errorMessage Validation error message

Promise with error result